homeassistant/components/teslemetry/device_tracker.py

```diff
@@ -0,0 +1,85 @@
+"""Device Tracker platform for Teslemetry integration."""
+
+from __future__ import annotations
+
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .entity import TeslemetryVehicleEntity
+from .models import TeslemetryVehicleData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Teslemetry device tracker platform from a config entry."""
+
+    async_add_entities(
+        klass(vehicle)
+        for klass in (
+            TeslemetryDeviceTrackerLocationEntity,
+            TeslemetryDeviceTrackerRouteEntity,
+        )
+        for vehicle in entry.runtime_data.vehicles
+    )
+
+
+class TeslemetryDeviceTrackerEntity(TeslemetryVehicleEntity, TrackerEntity):
+    """Base class for Teslemetry Tracker Entities."""
+
+    lat_key: str
+    lon_key: str
+
+    def __init__(
+        self,
+        vehicle: TeslemetryVehicleData,
+    ) -> None:
+        """Initialize the device tracker."""
+        super().__init__(vehicle, self.key)
+
+    def _async_update_attrs(self) -> None:
+        """Update the attributes of the device tracker."""
+
+        self._attr_available = (
+            self.get(self.lat_key, False) is not None
+            and self.get(self.lon_key, False) is not None
```

Comment:
I think if you want to check for `None`, you need to not have the default `False`
```suggestion
            self.get(self.lat_key) is not None
            and self.get(self.lon_key) is not None
```

Reply:
The default for get is None, but I am looking for a literal value of None, so I do need to set the default to literally anything other than None.

Reply:
Ok, so if I understand correctly:

```
self.get(self.lat_key, False) == 12.3456  # Tracker is available
self.get(self.lat_key, False) == None # Tracker is not available
self.get(self.lat_key, False) == False # Tracker is available???
```
I don't understand how the value can be missing, trigger the `get` default of False, but you want that scenario to be reported as 'available'. Have I misunderstood something?

Reply:
When a value is missing, its not unavailable, its unknown.

self.get(self.lat_key, False) == 12.3456  # Tracker is available
self.get(self.lat_key, False) == None # Tracker is not available
self.get(self.lat_key, False) == False # Tracker is unknown, but avaliable, likely because the vehicle is asleep when home assistant started.

Reply:
Ah ok, that makes sense now. I didn't realise there was an 'unknown', 'available' state. If that's what you need, looks good to me.